### ISSUE TITLE
performance concerns 

### ISSUE THREAD

**Issue by ledbit** [dcheckoway/cribl-checkoway#1]
Can you think of ways to improve performance in this block?
https://github.com/dcheckoway/cribl-checkoway/blob/c22aa24067bf849f2de3f3d0c3b772e11672ba7d/src/main/java/com/cribl/checkoway/FileScanner.java#L104

No need to implement it, comments would be sufficient.

**Comment by dcheckoway** [comment]
@ledbit  Definitely!  As you can tell, I went for simplicity here, using the JDK's built-in `BufferedReader.lines()` streaming mechanism.  Code readability won out over pure performance.

That said, if performance is paramount, we have the content sitting in `buf` and we could just read the bytes directly.  That might shave off a few nanoseconds.  😛 

Another way to improve performance would be to avoid the use of the `resultsForThisChunk` "staging" list entirely.  Building on what I mentioned in the previous comment about reading `buf` directly, we could scan it backward starting at the end -- this would allow us to find lines in reverse (most recent to least recent).  And then we could just add matching lines to `results` directly as we find matches...and short circuit out of this loop as soon as we found enough matches.

I think the combination of those two changes would improve performance from say 2ms per chunk to like 1.9ms per chunk.  😛   Again, I might lean toward code readability over ultimate performance, but hey...if low latency was make or break for the business, there's plenty of room to play here!!
